decred/decred/dcr/wire/msgversion.py

```diff
@@ -0,0 +1,317 @@
+"""
+Copyright (c) 2020, the Decred developers
+See LICENSE for details
+"""
+
+import time
+
+from decred import DecredError
+from decred.dcr.wire import netaddress, wire
+from decred.util.encode import ByteArray
+
+
+# MaxUserAgentLen is the maximum allowed length for the user agent field in a
+# version message (MsgVersion).
+MaxUserAgentLen = 256
+
+# DefaultUserAgent for wire in the stack
+DefaultUserAgent = "/tinywire:0.0.1/"
+
+# Command string for the MsgVersion.
+CmdVersion = "version"
+
+
+class MsgVersion:
+    """
+    MsgVersion implements the Message API and represents a Decred version
+    message.  It is used for a peer to advertise itself as soon as an outbound
+    connection is made.  The remote peer then uses this information along with
+    its own to negotiate.  The remote peer must then respond with a version
+    message of its own containing the negotiated values followed by a verack
+    message (MsgVerAck).  This exchange must take place before any further
+    communication is allowed to proceed.
+    """
+
+    def __init__(
+        self,
+        protocolVersion,
+        services,
+        timestamp,
+        addrYou,
+        addrMe,
+        nonce,
+        userAgent,
+        lastBlock,
+        disableRelayTx,
+    ):
+        """
+        Args:
+            protocolVersion (int): Version of the protocol the node is using.
+            services (int): Bitfield which identifies the enabled services.
```

Comment:
I wonder if the [IntFlag](https://docs.python.org/3.7/library/enum.html#intflag) enum type would be useful for bitfields.

Reply:
Seems unnecessary.